homeassistant/components/sensor/hydrawise.py

```diff
@@ -0,0 +1,72 @@
+"""
+Support for Hydrawise sprinkler.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/sensor.hydrawise/
+"""
+import logging
+
+import voluptuous as vol
+
+import homeassistant.helpers.config_validation as cv
+from homeassistant.components.hydrawise import (
+    DATA_HYDRAWISE, HydrawiseEntity, DEVICE_MAP, DEVICE_MAP_INDEX, SENSORS)
+from homeassistant.components.sensor import PLATFORM_SCHEMA
+from homeassistant.const import CONF_MONITORED_CONDITIONS
+
+DEPENDENCIES = ['hydrawise']
+
+_LOGGER = logging.getLogger(__name__)
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Optional(CONF_MONITORED_CONDITIONS, default=SENSORS):
+        vol.All(cv.ensure_list, [vol.In(SENSORS)]),
+})
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Set up a sensor for a Hydrawise device."""
+    hydrawise = hass.data[DATA_HYDRAWISE].data
+
+    sensors = []
+    for sensor_type in config.get(CONF_MONITORED_CONDITIONS):
+        for zone in hydrawise.relays:
+            sensors.append(HydrawiseSensor(zone, sensor_type))
+
+    add_devices(sensors, True)
+
+
+class HydrawiseSensor(HydrawiseEntity):
+    """A sensor implementation for Hydrawise device."""
+
+    @property
+    def state(self):
+        """Return the state of the sensor."""
+        return self._state
+
+    def update(self):
+        """Get the latest data and updates the states."""
+        mydata = self.hass.data['hydrawise'].data
+        _LOGGER.debug("Updating Hydrawise sensor: %s", self._name)
+        if self._sensor_type == 'watering_time':
+            if mydata.running is None or not mydata.running:
+                self._state = 0
+            else:
+                if int(mydata.running[0]['relay']) == self.data.get('relay'):
+                    self._state = int(mydata.running[0]['time_left']/60)
+                else:
+                    self._state = 0
+        else:  # _sensor_type == 'next_cycle'
+            for relay in mydata.relays:
+                if relay.get('relay') == self.data.get('relay'):
+                    if relay.get('nicetime') == 'Not scheduled':
+                        self._state = 'not_scheduled'
+                    else:
+                        self._state = relay.get('nicetime').split(',')[0] + \
```

Comment:
What time is this and how often does it change?

Reply:
This time (`nicetime`) indicates the next time the irrigation zone will run. If the user configures the Hydrawise cloud service to determine the next time the zone will run then `nicetime` is set by the cloud. Weather and soil conditions will dynamically change when the value of `nicetime`. 

If the user has HA turn on irrigation zones then `nicetime` will reflect when the zone will run under manual control.